### ISSUE TITLE
HackRF transmits okay first time, stops on second time, and okay third time, and repeats (add even). Why?

### ISSUE THREAD

**Issue by electrical-pro** [f4exb/sdrangel#1279]
https://user-images.githubusercontent.com/31592485/172065502-c15ec659-4b09-4887-8843-92f0582e70f2.mp4

Weird issue with HackRF One (please watch the video and listen).
When I press the start/stop button HackRF transmits on add times but not on even.
On odd times everything is fine.
On even times, there is a carrier, but no modulation and the carrier seems to be weaker.
I rarely use Hack RF for transmitting so I don't know when this issue started, maybe it's related to the firmware update (release 2021.03.1), but I'm not sure.

here are the settings I used
![image](https://user-images.githubusercontent.com/31592485/172065758-0bfc9871-26f4-4c0b-9bff-552ec115c535.png)
sdrangel-7.3.1

I also tested with old sdrangel-4.14.3 and it has the same issue.

I used sdrangel-4.14.3 a few years ago without any issues, so I assume it may be related to HackRF firmware update (release 2021.03.1) that I did a few months ago.

What is causing this issue?

**Comment by f4exb** [comment]
This is normal. HackRF has a residual carrier remaining at the last frequency it was set.

**Comment by electrical-pro** [comment]
@f4exb, I still do not fully understand. Why there is no modulation?
Maybe I explained the issue poorly or I don't understand something.
I expect HackRF to transmit a modulated signal every time it goes into TX mode.
But it only transmits modulated TX signal on odd times but not on even times.

Expected behaviour:
start/stop button clicked - transmission with modulation is ON
start/stop button clicked - transmission with modulation is OFF
start/stop button clicked - transmission with modulation is ON
start/stop button clicked - transmission with modulation is OFF
start/stop button clicked - transmission with modulation is ON
start/stop button clicked - transmission with modulation is OFF
start/stop button clicked - transmission with modulation is ON
start/stop button clicked - transmission with modulation is OFF

Observed behaviour:
start/stop button clicked - transmission with modulation is ON
start/stop button clicked - transmission with modulation is OFF
start/stop button clicked - transmission WITHOUT modulation is ON (why?)
start/stop button clicked - transmission WITHOUT modulation is OFF
start/stop button clicked - transmission with modulation is ON
start/stop button clicked - transmission with modulation is OFF
start/stop button clicked - transmission WITHOUT modulation is ON
start/stop button clicked - transmission WITHOUT modulation is OFF

**Comment by f4exb** [comment]
For me it works normally starting with modulation at each ON.

**Comment by electrical-pro** [comment]
@f4exb, may I ask if you using the latest HackRF firmware release 2021.03.1?

![image](https://user-images.githubusercontent.com/31592485/172139841-f3073c1f-c839-4386-a554-773d6a35cfce.png)

I tested on another computer with the same issue. So I assume it's somehow related to my HackRF, it doesn't look like a hardware issue, so I think the latest firmware causes this.

Can anyone confirm the issue on HackRF firmware release 2021.03.1?

**Comment by f4exb** [comment]
It is much older than that:
```
hackrf_info version: git-cc7f599                                                                                                                                                                                                            
libhackrf version: git-cc7f599 (0.5)                                                                                                                                                                                                        
Found HackRF                                                                                                                                                                                                                                
Index: 0                                                                                                                                                                                                                                    
Serial number: 0000000000000000088869dc3878a51b                                                                                                                                                                                             
Board ID Number: 2 (HackRF One)                                                                                                                                                                                                             
Firmware Version: 2017.02.1 (API:1.02)                                                                                                                                                                                                      
Part ID Number: 0xa000cb3c 0x006a4761   
```

**Comment by Radiomix2000** [comment]
> @f4exb, may I ask if you using the latest HackRF firmware release 2021.03.1?


Did you update CPLD too?

**Comment by electrical-pro** [comment]
Fixed the issue by updating to [release 2022.09.1](https://github.com/greatscottgadgets/hackrf/releases/tag/v2022.09.1)
